GVRf/Framework/jni/objects/animation/mesh_animation.cpp

```diff
@@ -0,0 +1,310 @@
+#include "mesh_animation.h"
+
+#include "util/gvr_log.h"
+
+#include <glm/glm.hpp>
+#include <glm/gtc/type_ptr.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+namespace gvr {
+
+void BoneData::addBoneData(int boneId, float boneWeight) {
+
+    for (int i = 0; i < sizeof(ids); i++) {
+        if (weights[i] == 0.0) {
+            ids[i] = boneId;
+            weights[i] = boneWeight;
+            return;
+        }
+    }
+}
+
+void MeshAnimation::setScene(const aiScene* scene) {
+    aiScene_ = scene;
```

Comment:
I would imagine all the Assimp-stuff will be confined to an importer unit/package that converts to our internal representation and that's it. The MeshAnimation is unusable without an aiScene. Are we supposed to support just the case where this stuff comes from importing a model? That would be ok as long as it is spelled out and could simplify certain things. The decision should be applicable to bones too then.

We could refactor these things later in case we want them in soon. If that is the case I will just stop making needless noise.